spirit_utils/src/ground_truth_publisher.cpp

```diff
@@ -29,16 +29,16 @@ void GroundTruthPublisher::mocapCallback(const geometry_msgs::PoseStamped::Const
 
   if (last_mocap_msg_ != NULL)
   {
-    // Collect change in position, and change in time for velocity update
+    // Collect change in position for velocity update
     double xDiff = msg->pose.position.x - last_mocap_msg_->pose.position.x;
     double yDiff = msg->pose.position.y - last_mocap_msg_->pose.position.y;
     double zDiff = msg->pose.position.z - last_mocap_msg_->pose.position.z;
-    double dt = spirit_utils::getROSMessageAgeInMs(msg->header) -
-                spirit_utils::getROSMessageAgeInMs(last_mocap_msg_->header);
 
-    mocap_vel_estimate_.x = alpha_*mocap_vel_estimate_.x + (1-alpha_)*xDiff/dt;
-    mocap_vel_estimate_.y = alpha_*mocap_vel_estimate_.y + (1-alpha_)*yDiff/dt;
-    mocap_vel_estimate_.z = alpha_*mocap_vel_estimate_.z + (1-alpha_)*zDiff/dt;
+    // Filtered velocity estimate assuming motion capture frame rate is constant at mocap_rate_
+    // in order to avoid variable network and ROS latency that appears in the message time stamp
+    mocap_vel_estimate_.x = alpha_*mocap_vel_estimate_.x + (1-alpha_)*xDiff*mocap_rate_;
+    mocap_vel_estimate_.y = alpha_*mocap_vel_estimate_.y + (1-alpha_)*yDiff*mocap_rate_;
+    mocap_vel_estimate_.z = alpha_*mocap_vel_estimate_.z + (1-alpha_)*zDiff*mocap_rate_;
```

Comment:
Are these supposed to be divided by mocap_rate rather than multiplied?

Reply:
Multiplying by the rate is equivalent to dividing by the sampling time right? This seems right to me.